beacon-chain/blockchain/process_block.go

```diff
@@ -508,14 +508,14 @@ func (s *Service) handleEpochBoundary(ctx context.Context, postState state.Beaco
 			return err
 		}
 	} else if postState.Slot() >= s.nextEpochBoundarySlot {
-		s.headLock.RLock()
-		st := s.head.state
-		s.headLock.RUnlock()
-		if err := reportEpochMetrics(ctx, postState, st); err != nil {
+		headSt, err := s.HeadState(ctx)
```

Comment:
Should we move epoch metric reporting to the end of this method? updating the caches seem more important.